### PR TITLE
Add `body_digest` option to `sign_v4/10`

### PR DESCRIPTION
**Pull request by wojtekmach**
An example where this is useful is: https://github.com/wojtekmach/req/pull/301 :)

### CODE REVIEW

**Comment by onno-vos-dev** [comment]
Will review, merge and release after work (in a couple hours) assuming everything looks good 👌

File: src/aws_signature.erl [hunk 1]
```diff
@@ -49,6 +49,12 @@ sign_v4(AccessKeyID, SecretAccessKey, Region, Service, DateTime, Method, URL, He
 %% this results in each segment being URI-encoded twice, as expected
 %% by AWS. Defaults to `true'.
 %% </dd>
+%% <dt>`body_digest'</dt>
+%% <dd>
+%% Optional SHA256 digest of the request body. This option can be used to provide
+%% a fixed digest value, such as "UNSIGNED-PAYLOAD", when sending requests without
+%% signing the body, <strong>which is expected for S3</strong>.
```
[the remark this answers is not in the document]
Comment (wojtekmach) [reply src/aws_signature.erl]: 😅

**Review by onno-vos-dev — APPROVED** [review]
Awaiting @jonatanklosko 👌

**Review by jonatanklosko — APPROVED** [review]

**Comment by onno-vos-dev** [comment]
Will release shortly 👌

**Comment by onno-vos-dev** [comment]
@jonatanklosko @wojtekmach I'll try to bring the release train PR https://github.com/aws-beam/aws_signature/pull/24 along in this change. So bear with me if it won't work right away, I'm on it 👌

**Comment by onno-vos-dev** [comment]
🎉 https://hex.pm/packages/aws_signature/0.3.2

**Comment by wojtekmach** [comment]
Thanks all, I really appreciate it!

**Pull request merged**